src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs

```diff
@@ -181,6 +183,11 @@ private MessageType GetMessageType(ReadOnlySpan<byte> line)
         {
             EnsureStartsWithDataPrefix(line);
 
+            if (line.Length != _messagTypeLineLength)
```

Comment:
I've moved these checks out of this method but I think that the length check should stay. 
Because of the check here https://github.com/aspnet/SignalR/blob/c0af8a3a258a5b0d5d338f7580bf842ad91ed1e1/src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs#L79 and here https://github.com/aspnet/SignalR/blob/c0af8a3a258a5b0d5d338f7580bf842ad91ed1e1/src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs#L45 we aready know that we end with a \r\n. 

Also to check the next character we would still need a reference to the length of the message type line. So we would still need something resemebling the _messageTypeLineLength.